hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -678,10 +678,15 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 				Type:               string(hyperv1.ValidReleaseImage),
 				ObservedGeneration: hcluster.Generation,
 			}
-			if err := r.validateReleaseImage(ctx, hcluster); err != nil {
+			err := r.validateReleaseImage(ctx, hcluster)
+			if err != nil {
 				condition.Status = metav1.ConditionFalse
 				condition.Message = err.Error()
-				condition.Reason = hyperv1.InvalidImageReason
+				if strings.Contains(err.Error(), "failed to get pull secret") {
```

Comment:
Strings messages are not a reliable contract, you need to `if apierrors.IsNotFound(err)` here